src/common/Collision/RegularGrid.h

```diff
@@ -73,61 +73,27 @@ class RegularGrid2D
 
     void insert(const T& value)
     {
-        G3D::Vector3 pos[9];
-        pos[0] = value.GetBounds().corner(0);
-        pos[1] = value.GetBounds().corner(1);
-        pos[2] = value.GetBounds().corner(2);
-        pos[3] = value.GetBounds().corner(3);
-        pos[4] = (pos[0] + pos[1]) / 2.0f;
-        pos[5] = (pos[1] + pos[2]) / 2.0f;
-        pos[6] = (pos[2] + pos[3]) / 2.0f;
-        pos[7] = (pos[3] + pos[0]) / 2.0f;
-        pos[8] = (pos[0] + pos[2]) / 2.0f;
-
-        NodeArray<Node> na;
-        for (uint8 i = 0; i < 9; ++i)
+        G3D::AABox bounds;
+        BoundsFunc::GetBounds(value, bounds);
+        Cell low = Cell::ComputeCell(bounds.low().x, bounds.low().y);
+        Cell high = Cell::ComputeCell(bounds.high().x, bounds.high().y);
+        for (int x = low.x; x <= high.x; ++x)
         {
-            Cell c = Cell::ComputeCell(pos[i].x, pos[i].y);
-            if (!c.isValid())
+            for (int y = low.y; y <= high.y; ++y)
             {
-                continue;
+                Node& node = getGrid(x, y);
+                node.insert(value);
+                memberTable.insert(MemberTable::value_type(&value, &node));
             }
-            Node& node = getGridFor(pos[i].x, pos[i].y);
-            na.AddNode(&node);
         }
-
-        for (uint8 i = 0; i < 9; ++i)
-        {
-            if (na._nodes[i])
-            {
-                na._nodes[i]->insert(value);
-            }
-            else
-            {
-                break;
-            }
-        }
-
-        memberTable.set(&value, na);
     }
 
     void remove(const T& value)
     {
-        NodeArray<Node>& na = memberTable[&value];
-        for (uint8 i = 0; i < 9; ++i)
-        {
-            if (na._nodes[i])
-            {
-                na._nodes[i]->remove(value);
-            }
-            else
-            {
-                break;
-            }
-        }
-
+        for (MemberTable::value_type& p : Acore::Containers::MapEqualRange(memberTable, &value))
```

Comment:
Tc has p set as auto. Maybe that would work? It would define it.
https://github.com/TrinityCore/TrinityCore/blob/e84447ae2aaab2ea58242a9d5d11c55516e7a22b/src/common/Collision/RegularGrid.h#L86

Reply:
will try

Reply:
seems to have worked, thanks